sanic/signals.py

```diff
@@ -131,3 +132,43 @@ def finalize(self, do_compile: bool = True):
             signal.ctx.event = asyncio.Event()
 
         return super().finalize(do_compile=do_compile)
+
+    def _build_event_parts(self, event: str) -> Tuple[str, str, str]:
+        parts = path_to_parts(event, self.delimiter)
+        if (
+            len(parts) != 3
+            or parts[0].startswith("<")
+            or parts[1].startswith("<")
+        ):
+            raise InvalidSignal("Invalid signal event: %s" % event)
+
+        current = Reservation(*parts)
+        for reservation in RESERVED_EVENTS:
+            if reservation.namespace == "*":
+                raise InvalidSignal(
+                    "Cannot declare reserved signal event: %s" % event
+                )
+
+            if reservation.namespace is not None and (
+                current.namespace in reservation.namespace.split(",")
+            ):
+                if (
+                    reservation.reference == "*"
+                    or current.reference
+                    not in reservation.reference.split(",")
+                ):
+                    raise InvalidSignal(
+                        "Cannot declare reserved signal event: %s" % event
+                    )
+
+                if reservation.reference is None or reservation.action is None:
+                    continue
```

Comment:
`reservation.reference` cannot be None here. We call `reservation.reference.split(",")` above, so it must be a string. It it was None Line 158 would throw and exception.
Do we need to move `if reservation.reference is None: continue` to above line 155?

Reply:
Good catch. Again, maybe I am overcomplicating and the whole thing should just be:

```python
parts = event.split(",")
if parts[0] in ("server", "http", "sanic"):
    raise InvalidSignal
```

And then worry about any more complicated logic when we get to it.

Reply:
Yep. I'd go with the simpler solution for now. We can think about reservation rules and how they are matched against signals, in the future.

Reply:
Okay, I updated it to only be reserved namespaces `http` and `server`.

I want to push people (via the docs) to get used to using `sanic.notice.something` when they have a simple use case and a full blown namespaced event doesn't make sense.